src/main/ts/ephox/photon/Reader.ts

```diff
@@ -15,15 +15,11 @@ var iframeDoc = function (element) {
   }
 };
 
-var doc = function (element) {
-  var optDoc = iframeDoc(element);
-  return optDoc.fold(function () {
-    return element;
-  }, function (v) {
-    return v;
-  });
+const doc = (element: Element<HTMLIFrameElement>) => {
+  const optDoc = iframeDoc(element);
+  return optDoc.fold(() => Traverse.owner(element), (v) => v);
```

Comment:
I think you could use `getOrThunk` here.

Reply:
Yeah we can, I didn't really look at those sorts of things, but I'll fix them up anyways.